### DISCUSSION TITLE
Navigating to the same route?

### DISCUSSION THREAD

**Discussion by abayomi185** [expo/router#506] [Q&A] [answered]
Hi, is it possible to navigate to the same route and add it to the navigation stack?

`router.push()` does not add to stack and seems to replace the current screen.

It seems that I'll need [React Navigation - getId](https://reactnavigation.org/docs/screen#getid) but this is not available in Expo Router's Stack Screen?

**Comment by abayomi185** [comment, marked as answer]
I have resolved my issue; adding a unique Id and using dynamic routes solves this.

**Reply by likeSo** [reply]
It will be navigating back to that same route?

**Reply by abayomi185** [reply]
Before yes, it seemed to replace the same screen.

Using dynamic routes seems to solve this. https://expo.github.io/router/docs/migration/react-navigation/screen/#getid
